eng/pipelines/update-azure-linux-pipeline.yml

```diff
@@ -0,0 +1,89 @@
+  # acquire a serverless agent to present the approval step, so the dev has to pay attention and
+  # catch it. Instead, let them approve it ahead of time by skipping it. The default is always used
+  # by automation, so this is only used for dev builds and retries.
+  - name: approveAheadOfTime
+    displayName: Approve right now, skipping the approval stage.
+    type: boolean
+    default: false
+
+  # Allow disabling specific parts of the internal build release. These can be used to re-run
+  # specific parts of a release if necessary, e.g. if a step silently failed.
+  - name: runCreatePullRequest
+    displayName: Create Pull Request on Azure Linux repo
+    type: boolean
+    default: true
+
+  # This parameter intentionally has no default: see release-go-pipeline.yml
+  - name: goReleaseConfigVariableGroup
+    displayName: '[Use "go-release-config" for a real release] Variable group that specifies release target locations and auth.'
+    type: string
+
+trigger: none
+pr: none
+
+variables:
+  - group: Microsoft-GoLang-bot
+  # Import config group. This may direct the build to use secrets from the other groups.
+  - group: ${{ parameters.goReleaseConfigVariableGroup }}
+  - name: assetJsonPath
+    value: $(Pipeline.Workspace)/build/BuildAssets/assets.json
+resources:
+  repositories:
+    - repository: 1ESPipelineTemplates
+      type: git
+      name: 1ESPipelineTemplates/1ESPipelineTemplates
+      ref: refs/tags/release
+
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    sdl:
+      sourceAnalysisPool:
+        name: NetCore1ESPool-Svc-Internal
+        image: 1es-ubuntu-2004
+        os: linux
+      suppression:
+        suppressionFile: $(Build.SourcesDirectory)/.config/guardian/.gdnsuppress
+      tsa:
+        enabled: true
+        configFile: $(Build.SourcesDirectory)/.config/tsa/tsaoptions.json
+
+    stages:
+      - ${{ if eq(parameters.approveAheadOfTime, false) }}:
+        - stage: WaitForApproval
+          jobs:
+            - job: Approve
+              pool: server
+              timeoutInMinutes: 1440 # 1 day
+              steps:
+                - task: ManualValidation@0
+                  inputs:
+                    instructions: >
+                      Once the microsoft/go build is complete, approve this step to automatically update 
+                      the Azure Linux specfiles and create a pull request.
+                    onTimeout: 'reject'
+
+      - stage: Release
+        jobs:
+          - template: /eng/pipelines/jobs/releasego.yml@self
+            parameters:
+              retryInstructionsFlags: '-preapproval'
+              steps:
+                # Resume the build at the furthest step based on the inputs. A fresh build has all 'nil', a
+                # retry build will have a non-nil value in one of these parameters. Only the value of the last
+                # step's parameter matters.
+                - download: build
```

Comment:
this is currently failing, I think we need to provide some more parameters such as the pipeline name etc

Reply:
Yeah, it should look like this:

https://github.com/microsoft/go-infra/blob/f0ee8dc8eb530988cbb8a971ce9290b8a9484704/eng/pipelines/steps/release-build-steps.yml#L155-L165

(Although here, it can use a parameter rather than variable for the run ID (build ID).)

To test it out, copy the number from the `buildId=###` part of the URL of a microsoft-go build.

Reply:
fixed